### PR TITLE
Support different guest program types

### PR DESCRIPTION
**Pull request by jsign**
The original motivation for this PR was to include _empty programs_ for benchmarking, as it's a known _base cost_ we want to understand better. 

In the near future, we might be interested in supporting other ad-hoc guest programs for benchmarking. For example, RLP/SSZ encoding blocks, hash functions benchmarks for binary trees, and potentially many more experiments. 

This motivated me to restructure the way guest programs are organized and run via `ere-hosts`.

The documentation was updated, but a TL;DR is:
- The `ere-guests` folder now contains one sub-folder per _guest program type_.
- For example, the previous guest programs now live in the `stateless-validator`, which was the only use case we had before. The internal structure of the guest program type folders is the same as before -- one folder per supported zkVM.
- I added now a new guest program type, `empty-program,` which was the original intention for this work.
- In the future, we can keep adding folders, which I believe nicely organize guest programs.
- `ere-hosts` create:
  - Now expects one argument indicating which guest program type you want to run. Supported values now are `stateless-validator` and `empty-program`.
  - I reorganized many things so the code is more organized, and the zkVM instance creation supports generic paths for target programs.

The main idea is that if we want to add a new guest program type, it should be easy. This is explained further in the docs.

I think this can also help by having a more complete CI to test most of the workspace mechanics, since we can now run the empty-program guest type as a lightweight target.  Improving the CI is planned in upcoming PRs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -116,25 +122,54 @@ This repository contains an `xtask` that will automate this process by calling `
...
-| **Zisk**         | `ere-guests/zisk/`    | `crates/ere-hosts/` | `zkevm-metrics/zisk/`      |
+| zkVM             | Stateless Validator Path                    | Empty Program Path                 | Host Path           | Metrics Output             |
+| ---------------- | ------------------------------------------- | ---------------------------------- | ------------------- | -------------------------- |
+| **Succinct SP1** | `ere-guests/stateless-validator/sp1/`       | `ere-guests/empty-program/sp1/`    | `crates/ere-hosts/` | `zkevm-metrics/sp1/`       |
+| **RISC0**        | `ere-guests/stateless-validator/risc0/`     | *Not implemented*                  | `crates/ere-hosts/` | `zkevm-metrics/risc0/`     |
+| **OpenVM**       | `ere-guests/stateless-validator/openvm/`    | *Not implemented*                  | `crates/ere-hosts/` | `zkevm-metrics/openvm/`    |
+| **Pico**         | `ere-guests/stateless-validator/pico/`      | *Not implemented*                  | `crates/ere-hosts/` | `zkevm-metrics/pico/`      |
+| **Zisk**         | `ere-guests/stateless-validator/zisk/`      | *Not implemented*                  | `crates/ere-hosts/` | `zkevm-metrics/zisk/`      |
```
[the remark this answers is not in the document]
Comment (kevaundray) [reply README.md:151]: Sounds good

File: crates/benchmark-runner/src/lib.rs [hunk 2]
```diff
@@ -29,35 +29,87 @@ pub enum Action {
...
+        }
+    };
+    let out_path = run_config
+        .output_folder
+        .join(format!("{zkvm_name}/empty_program.json"));
+
+    let report = BenchmarkRun {
+        name: "empty_program".to_string(),
+        timestamp_completed: zkevm_metrics::chrono::Utc::now(),
+        block_used_gas: 0,
```
[the remark this answers is not in the document]
Comment (kevaundray) [reply crates/benchmark-runner/src/lib.rs:77]: Yeah, I think this PR moves closer towards benchmarking general guest programs, see: https://github.com/eth-act/zkevm-benchmark-workload/issues/53

The meta field would need to not be filled in by the benchmark runner, ie it would need to be a component that calls the runner and then adds this extra data

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (kevaundray) [reply README.md:151]: sidenote: I had a thought to remove supported zkVM section, since it is kind of a pain to keep it updated all the time

**Review by kevaundray — APPROVED** [review]

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jsign) [reply README.md:151]: Agree, probably isn't needed. We can bundle removing it in any upcoming PR.
